File: app/src/main/java/com/bas/google_book_app/domain/dto/BookImageLinksDTO.java
package com.bas.google_book_app.domain.dto;

import com.google.gson.annotations.SerializedName;

import static com.bas.google_book_app.utilsdata.Constants.NO_THUMBNAIL_URL;

public class BookImageLinksDTO {

    @SerializedName("thumbnail")
    private String mThumbnail;

    @SerializedName("small")
    private String mThumbnailSmall;

    @SerializedName("medium")
    private String mThumbnailMedium;

    @SerializedName("large")
    private String mThumbnailLarge;

    public String getThumbnail() {
        if (mThumbnailLarge != null) return mThumbnailLarge;
        if (mThumbnailMedium != null) return mThumbnailMedium;
        if (mThumbnailSmall != null) return mThumbnailSmall;
        if (mThumbnail != null) return mThumbnail;
        return NO_THUMBNAIL_URL;
    }
}

File: app/src/main/java/com/bas/google_book_app/domain/dto/BookImageLinksDTO.kt
package com.bas.google_book_app.domain.dto

import com.bas.google_book_app.utilsdata.Constants
import com.google.gson.annotations.SerializedName

class BookImageLinksDTO {
    @SerializedName("thumbnail")
    private val mThumbnail: String? = null

    @SerializedName("small")
    private val mThumbnailSmall: String? = null

    @SerializedName("medium")
    private val mThumbnailMedium: String? = null

    @SerializedName("large")
    private val mThumbnailLarge: String? = null
    val thumbnail: String
        get() = (mThumbnailLarge
            ?: (mThumbnailMedium
                ?: (mThumbnailSmall ?: (mThumbnail ?: Constants.NO_THUMBNAIL_URL)))).toString()
}